### PR TITLE
Qt5 porting

### PR DESCRIPTION
**Pull request by cryos**

### CODE REVIEW

**Comment by cryos** [comment]
So this is my first stab at Qt 5 in the superbuild, using the qt5-porting branch in tomviz right now, but we can switch that back. Any obvious errors you see before we test this out @mathturtle @cquammen?

**Comment by mathturtle** [comment]
Before you merge, I'll need to update the buildbots to point to Qt5.  Right now all of them have a hardcoded qmake executable location that is a Qt4 qmake.

**Comment by cryos** [comment]
Sounds good

**Comment by mathturtle** [comment]
Looks fine to me.  I'll make the buildbot change when I'm done with the windows test build.

**Comment by mathturtle** [comment]
Does NOT work on Windows.

**Comment by mathturtle** [comment]
Qt fails to configure, something about 'configure.exe is not a recognized program'.  I'm working on a fix now.

**Comment by cquammen** [comment]
I believe qt5 has `configure.bat` instead of `configure.exe`.

**Comment by cryos** [comment]
Maybe we really do need the .zip variant, I can add that back, wasn't sure why PV's superbuild did that.

**Comment by mathturtle** [comment]
It does, but configure.bat is trying to call configure.exe which doesn't exist.  I think the issue is that we download the wrong package.  I'm trying the .zip variant now.

**Comment by cryos** [comment]
I suspect that is my fault, I didn't realize they were generating windows vs other OS source packages, seems surprising, but we can add in the if(WIN32) blah else() blah...

**Comment by mathturtle** [comment]
I'm testing it now.  The other issue I've found is that you need a short build directory name, otherwise things in Qt5's build tree exceed the Windows path length and builds fail there.

**Comment by cryos** [comment]
That is a well known issue, C:\tvb works well...

**Comment by mathturtle** [comment]
So I showed the Windows errors I was getting to @mathstuf this morning and he said "Oh yeah, I never did get Qt5 working on Windows in the superbuild.  If you get it to work, let me know."  This should be fun.

**Comment by cryos** [comment]
That would have been great to know, although why not just download the Qt provided Windows binaries for now, I don't really care if the superbuild can build Qt 5...

**Comment by cquammen** [comment]
They haven't historically posted 64-bit builds for windows, but there appears to be one for 5.5 available here:

http://www.qt.io/download-open-source/#section-2

File: CMakeLists.txt [hunk 1]
```diff
@@ -144,6 +148,10 @@ set(CM_PROJECTS_ALL)
 # Include all optional projects. These are the ones user can pick and choose.
 # Some of these allow using system libraries too.
 include(zlib)
+include(freetype)
+include(fontconfig)
```
Comment (cquammen) [inline CMakeLists.txt]: Is fontconfig needed on Apple? It is under projects/unix, which isn't added to the CMake module path on Apple.

Comment (mathturtle) [inline CMakeLists.txt]: Probably not.  I had to add an `if (NOT WIN32)` block around both fontconfig and freetype.  You could also add a dummy project in the main projects directory...

Comment (cryos) [inline CMakeLists.txt]: I can fix that, they are in the elseif (UNIX) block, simple fix.

Comment (mathturtle) [inline CMakeLists.txt]: From what Cory says you probably want `elseif(UNIX AND NOT APPLE)` for fontconfig.

Comment (cryos) [inline CMakeLists.txt]: I think the change I made in the commit previously pushed is fine, and I am currently building a test on bigmac of just Qt 5 that is proceeding as expected right now.

**Comment by cryos** [comment]
Verified that Qt 5 built on bigmac with the two additional commits appended yesterday. I think we are fine for building right now, and you can simply type 'make qt' in both cases to build a Qt only superbuild.

**Comment by cquammen** [comment]
Cool. I continued my local superbuild in the background and ParaView built fine. I ran into problems with the scipy installation not finding the built numpy installation, just FYI. I don't plan to debug further unless you indicate otherwise.

**Pull request merged**